packages/sw-lib/src/lib/sw-lib.js

```diff
@@ -351,6 +352,25 @@ class SWLib {
       requestWrapper: new RequestWrapper(wrapperOptions),
     });
   }
+
+  /**
+   * This method will register any default routes the library will need.
+   */
+  _registerDefaultRoutes() {
+    const revisionedManager = this._precacheManager.getRevisionedCacheManager();
+    const cacheFirstHandler = this.cacheFirst({
+      cacheName: revisionedManager.getCacheName(),
+    });
+
+    const route = new this.Route({
+      match: ({url, event}) => {
+        const cachedUrls = revisionedManager.getCachedUrls();
```

Comment:
Using the `install` event sounds fine.

An alternative might be to lazy-generate the list via a getter, and then return the same list subsequent times its called? Something like

https://github.com/GoogleChrome/sw-helpers/blob/beada35eca632b46459cf1a5a336f37f988de115/packages/sw-broadcast-cache-update/src/lib/behavior.js#L103

If there are usage scenarios in which we don't need the `getCachedUrls()` value then that approach could be slightly more efficient than doing it in the `install` handler unconditionally. But we're not talking about anything too significant, so feel free to implement it however you think is cleanest.